### ISSUE TITLE
[Ideas] - Settings Sync Roadmap

### ISSUE THREAD

**Issue by shanalikhan** [shanalikhan/code-settings-sync#1092]
I'm very happy to see that VS Code team is now implementing Settings Sync inside vscode that includes full end-to-end connectivity with backend provider.

Thats what Settings Sync is for, right!

Now that main implementation is being done inside visual studio code. 
We need to explore the extensibility support for Settings Sync.

I'm looking for ideas on how to improve and further take Settings Sync extension. 

Reference : https://github.com/microsoft/vscode/issues/2743#issuecomment-560116420

cc : @sandy081 , @kieferrm , @chrmarti

**Comment by marvingreenberg** [comment]
I'm using shan.code-settings-sync and mostly like it.  But, I find that I sometimes need/want to have two different syncs - like I set one set of keybindings/extensions/settings for working with a team, and another for personal projects of some different team.  So, maybe that the sync feature has the concept of multiple configured backends with a (maybe bottom status bar) w¥ of switching between them.  Also, maybe an explicit indicator of r/w vs r/o access to backends (for team lead vs peon)

I also would like for more configurability for extension sync, or maybe have it be a separate function.  I've occasionally synced-up a couple of extensions that I was just playing with -  some extensions are just for me and don't want them to sync up.

**Comment by shanalikhan** [comment]
>  So, maybe that the sync feature has the concept of multiple configured backends with a (maybe bottom status bar) w¥ of switching between them. Also, maybe an explicit indicator of r/w vs r/o access to backends (for team lead vs peon)

Yes , have a look on https://github.com/shanalikhan/code-settings-sync/issues/413 and https://github.com/shanalikhan/code-settings-sync/issues/935

For #935 , about 80% work is completed in `v3.4.4` branch. 

> I also would like for more configurability for extension sync, or maybe have it be a separate function

I would suggest you to look into the `v3.4.4` code , you can test it and improve it and let me know. I will release the next version for it, and then we will write classes for other back-end providers as well.
